librecad/src/actions/rs_actiondefault.cpp

```diff
@@ -109,16 +109,67 @@ void RS_ActionDefault::keyReleaseEvent(QKeyEvent* e) {
 }
 
 
+/*
+   Highlights hovered entities that are visible and not locked.
+
+   - by Melwyn Francis Carlo <carlo.melwyn@outlook.com>
+*/
+void RS_ActionDefault::highlightHoveredEntities(const RS_Vector& currentMousePosition)
+{
+    for (auto entity : *graphicView->getContainer())
+    {
+        if (entity != nullptr)
+        {
+            if (entity->isVisible() && ! entity->isLocked())
+            {
+                double hoverTolerance { 10.0 / graphicView->getFactor().magnitude() };
+
+                if (entity->rtti() == RS2::EntityEllipse)
```

Comment:
Don't know why the tolerance value for `RS2::EntityEllipse` is interpreted differently in the `RS_Ellipse::isPointOnEntity` function.

That is why I had to set the `10.0` factor to `1.0`, else it detects a far-off mouse position as being _on_ the elliptic entity.